crates/router/src/db/kafka_store.rs

```diff
@@ -2645,25 +2645,50 @@ impl UserRoleInterface for KafkaStore {
             .await
     }
 
-    async fn delete_user_role_by_user_id_merchant_id(
+    async fn list_user_roles_by_user_id(
         &self,
         user_id: &str,
-        merchant_id: &id_type::MerchantId,
         version: enums::UserRoleVersion,
-    ) -> CustomResult<user_storage::UserRole, errors::StorageError> {
+    ) -> CustomResult<Vec<user_storage::UserRole>, errors::StorageError> {
         self.diesel_store
-            .delete_user_role_by_user_id_merchant_id(user_id, merchant_id, version)
+            .list_user_roles_by_user_id(user_id, version)
             .await
     }
 
-    async fn list_user_roles_by_user_id(
+    async fn find_user_role_by_user_id_and_lineage(
         &self,
         user_id: &str,
+        org_id: &id_type::OrganizationId,
+        merchant_id: &id_type::MerchantId,
+        profile_id: Option<&String>,
         version: enums::UserRoleVersion,
-    ) -> CustomResult<Vec<user_storage::UserRole>, errors::StorageError> {
-        self.diesel_store
-            .list_user_roles_by_user_id(user_id, version)
-            .await
+    ) -> CustomResult<storage::UserRole, errors::StorageError> {
+        self.find_user_role_by_user_id_and_lineage(
+            user_id,
+            org_id,
+            merchant_id,
+            profile_id,
+            version,
+        )
+        .await
+    }
+
+    async fn delete_user_role_by_user_id_and_lineage(
+        &self,
+        user_id: &str,
+        org_id: &id_type::OrganizationId,
+        merchant_id: &id_type::MerchantId,
+        profile_id: Option<&String>,
+        version: enums::UserRoleVersion,
+    ) -> CustomResult<storage::UserRole, errors::StorageError> {
+        self.delete_user_role_by_user_id_and_lineage(
```

Comment:
```suggestion
        self.diesel_store.delete_user_role_by_user_id_and_lineage(
```

Reply:
In kakfaStore if we can self.functions will causes stack_overflow. So avoid following that.